### PR TITLE
Add canary related supporting infra

### PR DESCRIPTION
**Pull request by mbaijal**
### Description
Add canary related supporting infra same as [SageMaker Controller](https://github.com/aws-controllers-k8s/sagemaker-controller/pull/23) 
Just some basic changes were needed:
1. [In the Trust.json](https://github.com/aws-controllers-k8s/applicationautoscaling-controller/compare/main...mbaijal:canary-scripts?expand=1#diff-8aaa045d7219b24b3b596af797d2a6b4d4f19197264d3bf736412aa66de200f5R32)
2. [service name](https://github.com/aws-controllers-k8s/applicationautoscaling-controller/compare/main...mbaijal:canary-scripts?expand=1#diff-7607a45f8584cfc8941fccd0a49a65dea7d3949bea8d834af35471ea2ca93573R4)
3.  [CleanupCode](https://github.com/aws-controllers-k8s/applicationautoscaling-controller/compare/main...mbaijal:canary-scripts?expand=1#diff-21241d2da04cb8da691ca4ec4a8fa02205a138f1a26100ab67b93fda61fc73d4R31)

### Testing
Tested that there have been several successful runs on the internal code build project for 8+ hours ensuring proper cleanup.

### Test Logs 
```
[gw2] [ 33%] PASSED tests/test_scalabletarget.py::TestScalableTarget::test_smoke
--
144 | [gw0] [ 66%] PASSED tests/test_scalingpolicy.py::TestScalingPolicy::test_smoke
145 | [gw1] [100%] PASSED tests/test_sagemaker_endpoint_autoscaling.py::TestSageMakerEndpointAutoscaling::test_autoscaling
```

### CODE REVIEW

File: test/canary/scripts/install_controller_helm.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    local region="$3"
+    local namespace="$4"
+
+    yq w -i helm/values.yaml "serviceAccount.annotations" ""
+    yq w -i helm/values.yaml 'serviceAccount.annotations."eks.amazonaws.com/role-arn"' "$oidc_role_arn"
+    yq w -i helm/values.yaml "aws.region" $region
+
+    kubectl create namespace $namespace
+    kubectl apply -f helm/crds
+    helm install -n $namespace ack-$service-controller --skip-crds helm
```
Comment (RedbackThomson) [inline test/canary/scripts/install_controller_helm.sh:17]: `yq` writes can be replaced with `--set` when doing `helm install`.

e.g.

```bash
helm install -n $namespace ack-$service-controller --skip-crds --set "serviceAccount.annotations"="" --set 'serviceAccount.annotations."eks.amazonaws.com/role-arn"'="$oidc_role_arn" --set "aws.region"="$region" helm
```

File: test/canary/scripts/run_test.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+echo "Log helm charts are deployed properly"
+kubectl -n $NAMESPACE get pods
+kubectl get crds
+
+pushd $E2E_DIR
+  export PYTHONPATH=..
+  # create resources for test
+  python service_bootstrap.py
+  sleep 5m
```
Comment (RedbackThomson) [inline test/canary/scripts/run_test.sh:68]: Why?

**Review by akartsky — APPROVED** [review]

File: test/canary/scripts/run_test.sh (hunk 2 above)
Comment (akartsky) [reply test/canary/scripts/run_test.sh:68]: at the time of writing this script, the jobs sometimes weren't able to find s3 files due to no-blocking s3 file upload 
that's why sleep was necessary. 
Not sure if this issue is fixed now after you moved test code

File: test/canary/scripts/install_controller_helm.sh (hunk 1 above)
Comment (mbaijal) [reply test/canary/scripts/install_controller_helm.sh:17]: This is copy pasted as is from the sagemaker canary at the moment. It should be cleaned up and moved to a common repo (test-infra maybe). 
Would it be ok if we did this cleanup post preview and for GA on a larger scale ?

File: test/canary/scripts/run_test.sh (hunk 2 above)
Comment (mbaijal) [reply test/canary/scripts/run_test.sh:68]: I remember asking or at-least wondering the same thing, should not be needed but maybe @surajkota saw some issues, I will let him answer this one.

Comment (surajkota) [reply test/canary/scripts/run_test.sh:68]: yes, still need it

**Review by surajkota — APPROVED** [review]

File: test/canary/scripts/setup_oidc.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    {
+      "Effect": "Allow",
+      "Principal": {
+        "Federated": "arn:aws:iam::${account_id}:oidc-provider/${oidc_url}"
+      },
+      "Action": "sts:AssumeRoleWithWebIdentity",
+      "Condition": {
+        "StringEquals": {
+          "${oidc_url}:aud": "sts.amazonaws.com",
+          "${oidc_url}:sub": ["system:serviceaccount:${namespace}:ack-applicationautoscaling-controller"]
```
Comment (surajkota) [inline test/canary/scripts/setup_oidc.sh:32]: this can be made generic by using passing $SERVICE to this method

ok for now

**Pull request merged**